File: python/560_Subarray_Sum_Equals_K.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        prefixSumMap = {0: 1}
        total, count = 0, 0
        
        for num in nums:
            total += num
            
            if total - k in prefixSumMap:
                count += prefixSumMap[total - k]
            
            if total in prefixSumMap:
                prefixSumMap[total] += 1
            else:
                prefixSumMap[total] = 1
        
        return count
            
        
        
            

File: swift/560_subarray_sum_equals_k.swift
class Solution {
  func subarraySum(_ nums: [Int], _ k: Int) -> Int {
    var prefixSums: [Int: Int] = [0: 1]
    var curPrefixSum: Int = 0
    var count: Int = 0

    for num in nums {
      curPrefixSum += num

      let target: Int = curPrefixSum - k
      let subarrayCnt = prefixSums[target] != nil ? prefixSums[target]! : 0
      count += subarrayCnt
      prefixSums[curPrefixSum, default: 0] += 1
    }

    return count
  }
}
